fs/fs.go

```diff
@@ -117,14 +117,27 @@ func NewFsInfo(context Context) (FsInfo, error) {
 
 func (self *RealFsInfo) addLabels(context Context) {
 	dockerPaths := getDockerImagePaths(context)
+
+	// Determine if Docker is using devicemapper for its storage driver. If so, don't invoke
+	// self.updateDockerImagesPath below, as that will incorrectly override the device associated with
+	// LabelDockerImages (set previously up above in NewFsInfo).
+	dockerUsesDeviceMapper := false
+	for dev, p := range self.partitions {
+		if p.fsType == "devicemapper" && self.labels[LabelDockerImages] == dev {
```

Comment:
If we do that, we would get "true" for loop back too, wouldn't we?

On Thursday, January 21, 2016, Jimmi Dyson <notifications@github.com> wrote:

> In fs/fs.go
> <https://github.com/google/cadvisor/pull/1070#discussion_r50476917>:
>
> > @@ -117,14 +117,27 @@ func NewFsInfo(context Context) (FsInfo, error) {
> >
> >  func (self *RealFsInfo) addLabels(context Context) {
> >  	dockerPaths := getDockerImagePaths(context)
> > +
> > +	// Determine if Docker is using devicemapper for its storage driver. If so, don't invoke
> > +	// self.updateDockerImagesPath below, as that will incorrectly override the device associated with
> > +	// LabelDockerImages (set previously up above in NewFsInfo).
> > +	dockerUsesDeviceMapper := false
> > +	for dev, p := range self.partitions {
> > +		if p.fsType == "devicemapper" && self.labels[LabelDockerImages] == dev {
>
> Wouldn't this be better done using the StorageDriver value from docker
> info?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/1070/files#r50476917>.
>